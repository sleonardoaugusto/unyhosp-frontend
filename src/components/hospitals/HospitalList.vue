<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <router-link
      v-for="(h, i) in hospitals"
      :key="i"
      class="card col-md-3"
      :to="{name: 'hospital-utis', params: {hospitalId: h.id}}"
    >
      <div class="row mb-5">
        <p class="card__name">{{ h.name }}</p>
      </div>
      <div class="row justify-content-between w-100">
        <router-link :to="{name: 'attendanceRegister', params: {hospitalId: h.id}}"
                     class="card__action type-light-green">Novo atendimento
        </router-link>
        <span class="card__action type-yellow">Editar</span>
      </div>
    </router-link>
  </div>
</template>

<script>
import hospitals from '@/mocks/hospitals'

export default {
  name: 'HospitalList',
  data: () => ({
    hospitals: []
  }),
  created() {
    this.hospitals = hospitals.data
  }
}
</script>

<style scoped>

</style>