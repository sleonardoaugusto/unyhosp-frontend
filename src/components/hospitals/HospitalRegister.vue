<template>
  <div class="d-flex flex-wrap justify-content-center align-items-center flex-column">
    <div class="col-md-6 mb-3">
      <h2 class="type-h4 type-blue-smoke">Cadastrar Hospital</h2>
    </div>
    <form @submit.prevent="submit" class="col-md-6">
      <div class="form-group" :class="{ 'form-group--error': $v.hospital.name.$error }">
        <label class="form__label" for="HospitalName">Nome:</label>
        <input class="form__input" type="text" id="HospitalName" v-model.trim="$v.hospital.name.$model">
        <div v-if="!$v.hospital.name.required" class="error--message">Campo obrigat√≥rio</div>
      </div>
      <button class="button button-green" type="submit">CADASTRAR</button>
    </form>
  </div>
</template>

<script>
import HospitalService from '@/services/HospitalService'
import { required } from 'vuelidate/lib/validators'

export default {
  name: 'HospitalRegister',
  data: () => ({
    hospital: {
      name: ''
    }
  }),
  validations: {
    hospital: {
      name: { required }
    }
  },
  methods: {
    submit() {
      this.$v.$touch()
      if (!this.$v.$invalid) {
        HospitalService.post(this.hospital)
      }
    }
  }
}
</script>

<style scoped>

</style>