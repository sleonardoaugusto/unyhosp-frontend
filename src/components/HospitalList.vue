<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <router-link
      v-for="(h, i) in hospitals"
      :key="i"
      class="card col-md-3"
      :to="{name: 'hospital-utis', params: {hospitalId: h.id}}"
    >
      <div class="row mb-5">
        <p class="card__name">{{ h.name }}</p>
      </div>
      <div class="row justify-content-between w-100">
        <span class="card__action type-blue">Novo atendimento</span>
        <span class="card__action type-yellow">Editar</span>
      </div>
    </router-link>
  </div>
</template>

<script>
import hospitals from '@/mocks/hospitals'

export default {
  name: 'HospitalList',
  data: () => ({
    hospitals: []
  }),
  created() {
    this.hospitals = hospitals.data
  }
}
</script>

<style scoped>

</style>