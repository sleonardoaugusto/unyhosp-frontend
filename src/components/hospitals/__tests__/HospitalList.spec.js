import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import HospitalList from '@/components/hospitals/HospitalList'
import hospitals from '@/components/hospitals/__tests__/__mocks__/hospitals'

const localVue = createLocalVue()
localVue.use(VueRouter)

describe('HospitalList.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(HospitalList, {
      localVue
    })
  })

  it('component is rendering properly', () => {
    wrapper.setData({ hospitals: hospitals.data })
    expect(wrapper.html()).toMatchSnapshot()
  })
})