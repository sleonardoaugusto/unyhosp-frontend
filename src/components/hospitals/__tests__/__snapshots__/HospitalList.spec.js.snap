// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HospitalList.vue component is rendering properly 1`] = `
<div class="col-md-12 d-flex justify-content-center flex-wrap">
  <router-link-stub to="[object Object]" tag="a" event="click" class="card col-md-3">
    <div class="row mb-5">
      <p class="card__name">São Joaquim</p>
    </div>
    <div class="row justify-content-between w-100">
      <router-link-stub to="[object Object]" tag="a" event="click" class="card__action type-light-green">Novo atendimento
      </router-link-stub> <span class="card__action type-yellow">Editar</span>
    </div>
  </router-link-stub>
  <router-link-stub to="[object Object]" tag="a" event="click" class="card col-md-3">
    <div class="row mb-5">
      <p class="card__name">Albert Einstein</p>
    </div>
    <div class="row justify-content-between w-100">
      <router-link-stub to="[object Object]" tag="a" event="click" class="card__action type-light-green">Novo atendimento
      </router-link-stub> <span class="card__action type-yellow">Editar</span>
    </div>
  </router-link-stub>
  <router-link-stub to="[object Object]" tag="a" event="click" class="card col-md-3">
    <div class="row mb-5">
      <p class="card__name">São Pedro</p>
    </div>
    <div class="row justify-content-between w-100">
      <router-link-stub to="[object Object]" tag="a" event="click" class="card__action type-light-green">Novo atendimento
      </router-link-stub> <span class="card__action type-yellow">Editar</span>
    </div>
  </router-link-stub>
</div>
`;
